### ISSUE TITLE
update data which matched without set the id

### ISSUE THREAD

**Issue by istiakk** [elastic/elasticsearch-php#269]
i want to update multiple document, where it matches the query, but it always give this error....

{"error":"DocumentMissingException[[sprinkle][1] [content][_id]: document missing]","status":404}

this is my code---

        $aa = array('www.google.com, 'youtube.com, .............'')

        $doc = array();
        $doc['test_field'] = $aa;

        $params = array();
        $params['id'] = '_id';
        $params['index'] = 'my_index';
        $params['type'] = 'my_type';
        $params['body']['query']['bool']['must'] = array(
            array('match' => array('zipcode' => '1216')),
        );
        $params['id'] = '_id';
        $params['body'] = $doc;

        $params['body'] = array('doc' => $doc);
        $_REQUEST = $client->update($params);

Can you kindly tell me how i can fix this issue ...Thanks

**Comment by mod3st** [comment]
As I know, you can update your documents one by one only (or use bulk api to reduce the number of request).

1. Get documents by query (use scroll api when you have a lot of results).
2. Update them separately or with bulk api (after each scroll iteration).

**Comment by istiakk** [comment]
can you kindly provide an example to for the whole process, i a bit of confused

**Comment by polyfractal** [comment]
Yep, @mod3st is correct.

Currently you can only update documents if you know their explicit ID (either individually, or in bulk).  There is a [long-standing PR to implement an Update-By-Query](https://github.com/elastic/elasticsearch/issues/2230), but it has been stalled due to a lot of hairy technical problems.  

Tl;dr: doing a consistent query + update across distributed shards, when some of those shards may fail during the operation, is a very tricky problem.  And ES lacks a task-management capability right now (e.g. killing long running tasks like an update-by-query), which is also blocking progress.

Soo...I wouldn't expect to see the PR make progress any time soon, tbh. :)
